File: python/src/leetcode/1500-1749/1598.py
from typing import List

class Solution:
    def minOperations(self, logs: List[str]) -> int:
        count = 0

        for op in logs:
            match op:
                case "../":
                    if count > 0:
                        count -= 1
                case "./":
                    continue
                case _:
                    count += 1

        return count

File: cpp/1598.cpp
#include <string>

class Solution {
public:
  int minOperations(std::vector<std::string> &logs) {
    int acc = 0;

    for (auto op : logs) {
      if ((op == "../")) {
        if (acc > 0) {
          acc -= 1;
        }

      } else if (op == "./") {
        continue;
      } else {
        acc += 1;
      }
    }

    return acc;
  }
};